fdbserver/ApplyMetadataMutation.cpp

```diff
@@ -778,8 +799,11 @@ class ApplyMetadataMutationsImpl {
 				    .detail("PopVersion", popVersion)
 				    .detail("Tag", tag.toString())
 				    .detail("Server", decodeServerTagKey(kv.key));
-				logSystem->pop(popVersion, decodeServerTagValue(kv.value));
-				(*tag_popped)[tag] = popVersion;
+				if (!forResolver) {
+					logSystem->pop(popVersion, decodeServerTagValue(kv.value));
+					(*tag_popped)[tag] = popVersion;
+				}
+				ASSERT_WE_THINK(forResolver ^ (tag_popped != nullptr));
```

Comment:
Should this ASSERTION placed before the if statement, so that the nullptr is checked before accessing it?

Reply:
Makes sense. Done.